airbyte-integrations/connectors/source-s3/source_s3/v4/cursor.py

```diff
@@ -68,15 +68,17 @@ def _is_legacy_state(value: StreamState) -> bool:
             return False
         try:
             # Verify datetime format in history
-            item = list(value.get("history", {}).keys())[0]
-            datetime.strptime(item, Cursor._DATE_FORMAT)
+            history = value.get("history", {}).keys()
```

Comment:
Individually, `_is_legacy_state` is fine now but `set_initial_state` will raise `ValueError` because `_convert_legacy_state` will have an empty `converted_history`

Reply:
Synced on this - this case shouldn't happen but we will handle it if it does.